File: java/solutions/01-beginner/ex016_big_decimal_money/BigDecimalMoney.java
package fewolearning.exercises.beginner.ex016_big_decimal_money;

import java.math.BigDecimal;
import java.math.RoundingMode;

/*
Exercise 016 - BigDecimal money (reference solution).
*/
public final class BigDecimalMoney {
    private BigDecimalMoney() {
    }

    public static BigDecimal add(BigDecimal first, BigDecimal second) {
        return first.add(second);
    }

    public static BigDecimal roundToCents(BigDecimal amount, RoundingMode roundingMode) {
        return amount.setScale(2, roundingMode);
    }
}

File: kotlin/solutions/01-beginner/ex031_big_decimal_money/BigDecimalMoney.kt
package fewolearning.exercises.beginner.ex031_big_decimal_money

import java.math.BigDecimal
import java.math.RoundingMode

/*
Exercise 031 - BigDecimal money (reference solution).
*/
fun add(first: BigDecimal, second: BigDecimal): BigDecimal = first.add(second)

fun roundToCents(amount: BigDecimal): BigDecimal = amount.setScale(2, RoundingMode.HALF_UP)
